postprocessor/src/main/java/com/azure/autorest/postprocessor/util/PartialUpdateHandler.java

```diff
@@ -160,11 +164,140 @@ private static String handlePartialUpdateForClassOrInterfaceFile(String generate
+        NodeList<ModuleDirective> usesDirectiveNodeList = mergeDirectiveNodeList(usesDirectivesForGeneratedFile, usesDirectivesForExistingFile);
+        NodeList<ModuleDirective> providesDirectiveNodeList = mergeDirectiveNodeList(providesDirectivesForGeneratedFile, providesDirectivesForExistingFile);
+        addToEachTypeOfDirectiveList(directivesForGeneratedFile, requiresDirectivesForExistingFile, exportsDirectivesForExistingFile, opensDirectivesForExistingFile, usesDirectivesForExistingFile, providesDirectivesForExistingFile);
+
+        NodeList<ModuleDirective> moduleDirectives = new NodeList<>();
+        moduleDirectives.addAll(requiresDirectiveNodeList);
+        moduleDirectives.addAll(exportsDirectiveNodeList);
+        moduleDirectives.addAll(opensDirectiveNodeList);
+        moduleDirectives.addAll(usesDirectiveNodeList);
+        moduleDirectives.addAll(providesDirectiveNodeList);
+
+        ModuleDeclaration moduleDeclaration = compilationUnitForGeneratedFile.getModule().get();
+        moduleDeclaration.setDirectives(moduleDirectives);
+
+        compilationUnitForGeneratedFile.setModule(moduleDeclaration);
+
+        return compilationUnitForGeneratedFile.toString();
+    }
+
+    private static NodeList<ModuleDirective> mergeDirectiveNodeList(NodeList<ModuleDirective> list1, NodeList<ModuleDirective> list2) {
+        NodeList<ModuleDirective> res = new NodeList<>();
+        res.addAll(list1);
+        for (ModuleDirective directive2 : list2) {
+            boolean isInList1 = false;
+            for (ModuleDirective directive1 : list1) {
+                if (directive1.getTokenRange().isPresent() && directive2.getTokenRange().isPresent()) {
+                    List<JavaToken> tokenList1 = new ArrayList<>();
+                    List<JavaToken> tokenList2 = new ArrayList<>();
+                    for (JavaToken token1 : directive1.getTokenRange().get()) {
+                        if (!token1.asString().trim().isEmpty()) {
+                            tokenList1.add(token1);
+                        }
+                    }
+                    for (JavaToken token2 : directive2.getTokenRange().get()) {
+                        if (!token2.asString().trim().isEmpty()) {
+                            tokenList2.add(token2);
+                        }
+                    }
+                    if (tokenList1.size() == tokenList2.size()) {
+                        boolean isTokenList1EqualsTokenList2 = true;
+                        for (int i = 0; i < tokenList1.size(); ++i) {
+                            if (!tokenList1.get(i).asString().trim().equals(tokenList2.get(i).asString().trim())) {
+                                isTokenList1EqualsTokenList2 = false;
+                                break;
+                            }
+                        }
+                        isInList1 = isTokenList1EqualsTokenList2;
+                    }
```

Comment:
At least write some comments on what you are doing here.

It is maybe a hack that works in most times. I don't like it but I can accept it.

@srnagar Let us know if you have opinion on this.

Reply:
Added Javadoc and comments on the logic